packages/@aws-cdk/aws-appmesh/lib/backend.ts

```diff
@@ -0,0 +1,92 @@
+
+import { CfnVirtualNode } from './appmesh.generated';
+import { ClientPolicy } from './client-policy';
+import { IVirtualService } from './virtual-service';
+
+// keep this import separate from other imports to reduce chance for merge conflicts with v2-main
+// eslint-disable-next-line no-duplicate-imports, import/order
+import { Construct } from '@aws-cdk/core';
+
+/**
+ * Represents the properties needed to define backend defaults
+ */
+export interface BackendDefaultsOptions {
+  /**
+   * Client policy for backend defaults
+   *
+   * @default none
+   */
+  readonly clientPolicy?: ClientPolicy;
+}
+
+/**
+ * Represents the properties needed to define a backend
+ */
+export interface BackendOptions {
+  /**
+   * The Virtual Service this backend points to
+   */
+  readonly virtualService: IVirtualService;
+  /**
+   * Client policy for a backend
+   *
+   * @default none
+   */
+  readonly clientPolicy?: ClientPolicy;
+}
+
+/**
+ * Provides static factory methods to generate backend API structures
+ */
+export class Backends {
+  /**
+   * Creates a backend defaults
+   */
+  public static backendDefaults(props: BackendDefaultsOptions): BackendDefaults {
+    return new BackendDefaults(props.clientPolicy);
+  }
+  /**
+   * Creates a named backend
+   */
+  public static backend(props: BackendOptions): Backend {
+    return new Backend(props.virtualService, props.clientPolicy);
+  }
+}
+
+/**
+ * Represents all the backends that aren't specifically defined using the backend .
+ */
+export class BackendDefaults {
```

Comment:
I wonder if we could just make this an interface. I'm trying to think of a situation where we would need to enforce mutual exclusivity for this field and I'm not sure if we would need this. What do you think?

Reply:
Not sure if I'm following. Do you mean both `Backend` and `BackendDefaults` would implement an interface that enforces a method to get the client policy?

Reply:
Like, just defining BackendDefaults as an interface type

```typescript
interface BackendDefaults {
      readonly clientPolicy?: ClientPolicy;
}
```

and then instead of having to call `Backends.backendDefaults` you could just create a new object like:


```typescript
new VirtualNode(stack, 'VirtualNode', {
    ...
    backendDefaults: {
        clientPolicy: appmesh.ClientPolicy.acm({ ... }),
    }
);
```

Reply:
My understanding is the `bind()` methods are used to enforce exclusivity. Like you cannot define both `file` and `acm` types for ClientPolicies. For backend defaults, I don't see us needing to enforce this mutual exclusivity

Reply:
I would hesitate to remove to remove the `bind()` pattern for the backend itself. I could see there being other targets in the future that might not be a `IVirtualService`. We don't have any immediate plans for this, but adding a different type here is within the realm of reason.

Reply:
Fair enough. I defer to the expert(s) here 🙂.